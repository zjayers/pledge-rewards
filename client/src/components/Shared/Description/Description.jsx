import React from 'react';
import { DescCollapseBtn, DescriptionContainer } from './Description.style';

const Description = (props) => {
  const { description, descOpen, activated, setDescOpen } = props;
  let { descClicked } = props;

  if (!descClicked) {
    descClicked = () => setDescOpen(!descOpen);
  }

  /**
   * Render Description Button
   * Renders the 'Read more...' or 'Less' button depending on if the description box is open or collapsed
   * @returns {string|*}
   */
  const renderDescButton = () => {
    if (activated) {
      return descOpen ? 'Less' : 'Read more...';
    }

    /* Return a Fragment */
    return <></>;
  };

  return (
    <DescriptionContainer open={descOpen} activated={activated}>
      <p>{description}</p>
      <DescCollapseBtn onClick={descClicked} open={descOpen}>
        {renderDescButton()}
      </DescCollapseBtn>
    </DescriptionContainer>
  );
};

export default Description;
