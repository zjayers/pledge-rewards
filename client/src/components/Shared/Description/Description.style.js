import styled from 'styled-components';

export const DescriptionContainer = styled.div`
  font-size: 1.2rem;
  line-height: 1.5;
  position: relative;
  height: ${(props) => {
    if (props.activated) {
      return props.open ? 'auto' : '40px';
    }

    return 'auto';
  }};
  color: ${(props) => props.theme.colors.flavorText};
  display: grid;
  grid-template-rows: ${(props) => (props.activated ? '1fr 20px;' : '1fr')};
  overflow: hidden;
  margin-bottom: ${(props) => (props.open ? '1rem' : '20px')};
`;

export const DescCollapseBtn = styled.p`
  color: ${(props) => props.theme.colors.text};
  background: linear-gradient(
    to right,
    rgba(255, 255, 255, 0) 0%,
    rgba(255, 255, 255, 1) 20%
  );
  width: 80px;
  text-align: end;
  cursor: pointer;
  position: absolute;
  right: 5px;
  bottom: 0;
`;
