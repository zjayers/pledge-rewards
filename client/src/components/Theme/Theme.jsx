import React from 'react';
import { ThemeProvider } from 'styled-components';

const theme = {
  colors: {
    borderGrey: '#9B9E9E',
    highlightGreen: '#009E74',
    buttonGreen: '#009E74',
    gradientBlue: '#DBE7FE',
    gradientOrange: '#FFF2EC',
    text: '#282828',
    flavorText: '#888B8B',
    linkText: '#498F80',
  },
  fonts: ['Muli', 'sans-serif'],
  fontSizes: {
    small: '1rem',
    medium: '1.8rem',
    large: '3rem',
  },
};

const Theme = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

export default Theme;
