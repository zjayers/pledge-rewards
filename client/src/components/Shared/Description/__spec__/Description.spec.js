import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Description from '../Description';
import Theme from '../../../Theme/Theme';

describe('Description Component', () => {
  let component;
  let componentV2;
  let componentV3;

  beforeEach(() => {
    component = mount(
      <Theme>
        <Description activated descOpen descClicked={() => {}} />
      </Theme>
    );

    componentV2 = mount(
      <Theme>
        <Description activated={false} descOpen={false} />
      </Theme>
    );

    componentV3 = mount(
      <Theme>
        <Description activated descOpen={false} />
      </Theme>
    );
  });

  afterEach(() => {
    component.unmount();
    componentV2.unmount();
    componentV3.unmount();
  });

  it('should render without failure', () => {
    expect(component).toBeDefined();
  });

  it('should match the test snapshot', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should return an empty fragment if the "activated" prop is false', () => {
    expect(toJson(componentV2)).toMatchSnapshot();
  });

  it('should return an "Read More..." if the "activated" prop is true and the "descOpen" prop is false', () => {
    expect(toJson(componentV3)).toMatchSnapshot();
  });
});
