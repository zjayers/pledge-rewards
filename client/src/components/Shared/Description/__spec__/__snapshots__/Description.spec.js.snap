// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description Component should match the test snapshot 1`] = `
<Theme>
  <ThemeProvider
    theme={
      Object {
        "colors": Object {
          "borderGrey": "#9B9E9E",
          "buttonGreen": "#009E74",
          "flavorText": "#888B8B",
          "gradientBlue": "#DBE7FE",
          "gradientOrange": "#FFF2EC",
          "highlightGreen": "#009E74",
          "linkText": "#498F80",
          "text": "#282828",
        },
        "fontSizes": Object {
          "large": "3rem",
          "medium": "1.8rem",
          "small": "1rem",
        },
        "fonts": Array [
          "Muli",
          "sans-serif",
        ],
      }
    }
  >
    <Description
      activated={true}
      descClicked={[Function]}
      descOpen={true}
    >
      <Descriptionstyle__DescriptionContainer
        activated={true}
        open={true}
      >
        <div
          className="Descriptionstyle__DescriptionContainer-wtqy6i-0 iZdCHd"
          open={true}
        >
          <p />
          <Descriptionstyle__DescCollapseBtn
            onClick={[Function]}
            open={true}
          >
            <p
              className="Descriptionstyle__DescCollapseBtn-wtqy6i-1 loBJcu"
              onClick={[Function]}
              open={true}
            >
              Less
            </p>
          </Descriptionstyle__DescCollapseBtn>
        </div>
      </Descriptionstyle__DescriptionContainer>
    </Description>
  </ThemeProvider>
</Theme>
`;

exports[`Description Component should return an "Read More..." if the "activated" prop is true and the "descOpen" prop is false 1`] = `
<Theme>
  <ThemeProvider
    theme={
      Object {
        "colors": Object {
          "borderGrey": "#9B9E9E",
          "buttonGreen": "#009E74",
          "flavorText": "#888B8B",
          "gradientBlue": "#DBE7FE",
          "gradientOrange": "#FFF2EC",
          "highlightGreen": "#009E74",
          "linkText": "#498F80",
          "text": "#282828",
        },
        "fontSizes": Object {
          "large": "3rem",
          "medium": "1.8rem",
          "small": "1rem",
        },
        "fonts": Array [
          "Muli",
          "sans-serif",
        ],
      }
    }
  >
    <Description
      activated={true}
      descOpen={false}
    >
      <Descriptionstyle__DescriptionContainer
        activated={true}
        open={false}
      >
        <div
          className="Descriptionstyle__DescriptionContainer-wtqy6i-0 hRNJRj"
          open={false}
        >
          <p />
          <Descriptionstyle__DescCollapseBtn
            onClick={[Function]}
            open={false}
          >
            <p
              className="Descriptionstyle__DescCollapseBtn-wtqy6i-1 loBJcu"
              onClick={[Function]}
              open={false}
            >
              Read more...
            </p>
          </Descriptionstyle__DescCollapseBtn>
        </div>
      </Descriptionstyle__DescriptionContainer>
    </Description>
  </ThemeProvider>
</Theme>
`;

exports[`Description Component should return an empty fragment if the "activated" prop is false 1`] = `
<Theme>
  <ThemeProvider
    theme={
      Object {
        "colors": Object {
          "borderGrey": "#9B9E9E",
          "buttonGreen": "#009E74",
          "flavorText": "#888B8B",
          "gradientBlue": "#DBE7FE",
          "gradientOrange": "#FFF2EC",
          "highlightGreen": "#009E74",
          "linkText": "#498F80",
          "text": "#282828",
        },
        "fontSizes": Object {
          "large": "3rem",
          "medium": "1.8rem",
          "small": "1rem",
        },
        "fonts": Array [
          "Muli",
          "sans-serif",
        ],
      }
    }
  >
    <Description
      activated={false}
      descOpen={false}
    >
      <Descriptionstyle__DescriptionContainer
        activated={false}
        open={false}
      >
        <div
          className="Descriptionstyle__DescriptionContainer-wtqy6i-0 lhnXnZ"
          open={false}
        >
          <p />
          <Descriptionstyle__DescCollapseBtn
            onClick={[Function]}
            open={false}
          >
            <p
              className="Descriptionstyle__DescCollapseBtn-wtqy6i-1 loBJcu"
              onClick={[Function]}
              open={false}
            />
          </Descriptionstyle__DescCollapseBtn>
        </div>
      </Descriptionstyle__DescriptionContainer>
    </Description>
  </ThemeProvider>
</Theme>
`;
